### DISCUSSION TITLE
Uploading project to public github without the database password being revoked

### DISCUSSION THREAD

**Discussion by markslvk** [planetscale/discussion#496] [Q&A] [answered]
hello. How are you? I have a python project that uses planetScale as a database hosting, and i want to upload it as a public github for college, but when I upload it I receive this message: database password has been detected in a public repository on GitHub. This password has been revoked.

How can I bypass this?

**Comment by mscoutermarsh** [comment, marked as answer]
Hi @markslvk,

You'll want to update your code to use environment variables and store the password as a secret. Where you store it will depend on how the application is deployed. For example, if using Heroku, you'd follow [these directions to set them](https://devcenter.heroku.com/articles/config-vars).

If you're sharing the project with someone else, you could put the secrets in a .env file. You'd then ignore this .env file in git so it isn't checked into the repo.

The reason why the password gets revoked is because there are bad actors who continuously scrape public GitHub for credentials. By publishing the password to a public repo, this gives them access to your database.
